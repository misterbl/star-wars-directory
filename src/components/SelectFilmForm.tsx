import React, { PureComponent } from "react";
import { FormikProps } from "formik";
import moviesList from "../const/moviesList";

export interface ISelectFilmForm {
  selectedFilm: string;
}

class SelectFilmForm extends PureComponent<FormikProps<ISelectFilmForm>> {
  onSelectChange = async (event: React.ChangeEvent<HTMLSelectElement>) => {
    const { setFieldValue, handleSubmit } = this.props;
    await setFieldValue("selectedFilm", event.target.value);
    handleSubmit();
  };

  render() {
    return (
      <form onSubmit={this.props.handleSubmit} className="form-group mb-0">
        <select
          className="custom-select"
          style={{ width: "200px" }}
          onChange={this.onSelectChange}
        >
          <option value="">See all</option>
          {moviesList.map(({ value, text }) => (
            <option key={value} value={value}>
              {text}
            </option>
          ))}
        </select>
      </form>
    );
  }
}
export default SelectFilmForm;
