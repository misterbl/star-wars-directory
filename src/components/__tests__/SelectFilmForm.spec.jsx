import * as React from "react";
import { shallow } from "enzyme";
import SelectFilmForm from "../SelectFilmForm";

describe("SelectFilmForm", () => {
  const props = {
    setFieldValue: jest.fn(),
    values: {
      selectedFilm: "selectedFilm"
    },
    handleSubmit: jest.fn()
  };
  const wrapper = shallow(<SelectFilmForm {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls setFieldValue on input change", () => {
    const select = wrapper.find("select");
    select.simulate("change", { target: { value: "value" } });
    expect(props.setFieldValue).toHaveBeenCalledWith("selectedFilm", "value");
  });
  it("calls handleSubmit on form submit", () => {
    const form = wrapper.find("form");
    form.simulate("submit");
    expect(props.handleSubmit).toHaveBeenCalled();
  });
});
