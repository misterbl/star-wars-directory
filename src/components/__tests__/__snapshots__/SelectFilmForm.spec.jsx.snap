// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilmForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectFilmForm
    handleSubmit={[MockFunction]}
    setFieldValue={[MockFunction]}
    values={
      Object {
        "selectedFilm": "selectedFilm",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        className="custom-select"
        onChange={[Function]}
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <option
          value=""
        >
          See all
        </option>
        <option
          value="https://swapi.co/api/films/1/"
        >
          A New Hope
        </option>
        <option
          value="https://swapi.co/api/films/2/"
        >
          The Empire Strikes Back
        </option>
        <option
          value="https://swapi.co/api/films/3/"
        >
          Return of the Jedi
        </option>
        <option
          value="https://swapi.co/api/films/4/"
        >
          The Phantom Menace
        </option>
        <option
          value="https://swapi.co/api/films/5/"
        >
          Attack of the Clones
        </option>
        <option
          value="https://swapi.co/api/films/6/"
        >
          Revenge of the Sith
        </option>
        <option
          value="https://swapi.co/api/films/7/"
        >
          The Force Awakens
        </option>
      </select>,
      "className": "form-group mb-0",
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value=""
          >
            See all
          </option>,
          Array [
            <option
              value="https://swapi.co/api/films/1/"
            >
              A New Hope
            </option>,
            <option
              value="https://swapi.co/api/films/2/"
            >
              The Empire Strikes Back
            </option>,
            <option
              value="https://swapi.co/api/films/3/"
            >
              Return of the Jedi
            </option>,
            <option
              value="https://swapi.co/api/films/4/"
            >
              The Phantom Menace
            </option>,
            <option
              value="https://swapi.co/api/films/5/"
            >
              Attack of the Clones
            </option>,
            <option
              value="https://swapi.co/api/films/6/"
            >
              Revenge of the Sith
            </option>,
            <option
              value="https://swapi.co/api/films/7/"
            >
              The Force Awakens
            </option>,
          ],
        ],
        "className": "custom-select",
        "onChange": [Function],
        "style": Object {
          "width": "200px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "See all",
            "value": "",
          },
          "ref": null,
          "rendered": "See all",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/1/",
          "nodeType": "host",
          "props": Object {
            "children": "A New Hope",
            "value": "https://swapi.co/api/films/1/",
          },
          "ref": null,
          "rendered": "A New Hope",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/2/",
          "nodeType": "host",
          "props": Object {
            "children": "The Empire Strikes Back",
            "value": "https://swapi.co/api/films/2/",
          },
          "ref": null,
          "rendered": "The Empire Strikes Back",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/3/",
          "nodeType": "host",
          "props": Object {
            "children": "Return of the Jedi",
            "value": "https://swapi.co/api/films/3/",
          },
          "ref": null,
          "rendered": "Return of the Jedi",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/4/",
          "nodeType": "host",
          "props": Object {
            "children": "The Phantom Menace",
            "value": "https://swapi.co/api/films/4/",
          },
          "ref": null,
          "rendered": "The Phantom Menace",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/5/",
          "nodeType": "host",
          "props": Object {
            "children": "Attack of the Clones",
            "value": "https://swapi.co/api/films/5/",
          },
          "ref": null,
          "rendered": "Attack of the Clones",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/6/",
          "nodeType": "host",
          "props": Object {
            "children": "Revenge of the Sith",
            "value": "https://swapi.co/api/films/6/",
          },
          "ref": null,
          "rendered": "Revenge of the Sith",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "https://swapi.co/api/films/7/",
          "nodeType": "host",
          "props": Object {
            "children": "The Force Awakens",
            "value": "https://swapi.co/api/films/7/",
          },
          "ref": null,
          "rendered": "The Force Awakens",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          className="custom-select"
          onChange={[Function]}
          style={
            Object {
              "width": "200px",
            }
          }
        >
          <option
            value=""
          >
            See all
          </option>
          <option
            value="https://swapi.co/api/films/1/"
          >
            A New Hope
          </option>
          <option
            value="https://swapi.co/api/films/2/"
          >
            The Empire Strikes Back
          </option>
          <option
            value="https://swapi.co/api/films/3/"
          >
            Return of the Jedi
          </option>
          <option
            value="https://swapi.co/api/films/4/"
          >
            The Phantom Menace
          </option>
          <option
            value="https://swapi.co/api/films/5/"
          >
            Attack of the Clones
          </option>
          <option
            value="https://swapi.co/api/films/6/"
          >
            Revenge of the Sith
          </option>
          <option
            value="https://swapi.co/api/films/7/"
          >
            The Force Awakens
          </option>
        </select>,
        "className": "form-group mb-0",
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              See all
            </option>,
            Array [
              <option
                value="https://swapi.co/api/films/1/"
              >
                A New Hope
              </option>,
              <option
                value="https://swapi.co/api/films/2/"
              >
                The Empire Strikes Back
              </option>,
              <option
                value="https://swapi.co/api/films/3/"
              >
                Return of the Jedi
              </option>,
              <option
                value="https://swapi.co/api/films/4/"
              >
                The Phantom Menace
              </option>,
              <option
                value="https://swapi.co/api/films/5/"
              >
                Attack of the Clones
              </option>,
              <option
                value="https://swapi.co/api/films/6/"
              >
                Revenge of the Sith
              </option>,
              <option
                value="https://swapi.co/api/films/7/"
              >
                The Force Awakens
              </option>,
            ],
          ],
          "className": "custom-select",
          "onChange": [Function],
          "style": Object {
            "width": "200px",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "See all",
              "value": "",
            },
            "ref": null,
            "rendered": "See all",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/1/",
            "nodeType": "host",
            "props": Object {
              "children": "A New Hope",
              "value": "https://swapi.co/api/films/1/",
            },
            "ref": null,
            "rendered": "A New Hope",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/2/",
            "nodeType": "host",
            "props": Object {
              "children": "The Empire Strikes Back",
              "value": "https://swapi.co/api/films/2/",
            },
            "ref": null,
            "rendered": "The Empire Strikes Back",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/3/",
            "nodeType": "host",
            "props": Object {
              "children": "Return of the Jedi",
              "value": "https://swapi.co/api/films/3/",
            },
            "ref": null,
            "rendered": "Return of the Jedi",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/4/",
            "nodeType": "host",
            "props": Object {
              "children": "The Phantom Menace",
              "value": "https://swapi.co/api/films/4/",
            },
            "ref": null,
            "rendered": "The Phantom Menace",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/5/",
            "nodeType": "host",
            "props": Object {
              "children": "Attack of the Clones",
              "value": "https://swapi.co/api/films/5/",
            },
            "ref": null,
            "rendered": "Attack of the Clones",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/6/",
            "nodeType": "host",
            "props": Object {
              "children": "Revenge of the Sith",
              "value": "https://swapi.co/api/films/6/",
            },
            "ref": null,
            "rendered": "Revenge of the Sith",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "https://swapi.co/api/films/7/",
            "nodeType": "host",
            "props": Object {
              "children": "The Force Awakens",
              "value": "https://swapi.co/api/films/7/",
            },
            "ref": null,
            "rendered": "The Force Awakens",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
